### PR TITLE
Next: Use `<<package>>::<<import>>` naming convention for mock names

### PR DESCRIPTION
**Pull request by kasperpeulen**
## What I did

Use a new naming convention to make actions easier to scan

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This pull request has been released as version `0.0.0-pr-26853-sha-8dbd04c4`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-26853-sha-8dbd04c4 sandbox` or in an existing project with `npx storybook@0.0.0-pr-26853-sha-8dbd04c4 upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-26853-sha-8dbd04c4`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-26853-sha-8dbd04c4) |
| **Triggered by** | @kasperpeulen |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`kasper/mock-names`](https://github.com/storybookjs/storybook/tree/kasper/mock-names) |
| **Commit** | [`8dbd04c4`](https://github.com/storybookjs/storybook/commit/8dbd04c4c73b6a55fc9e77b129435fc8fde0e7be) |
| **Datetime** | Tue Apr 16 13:10:18 UTC 2024 (`1713273018`) |
| **Workflow run** | [8706234363](https://github.com/storybookjs/storybook/actions/runs/8706234363) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=26853`_
</details>

### CODE REVIEW

File: code/frameworks/nextjs/src/export-mocks/navigation/index.ts [hunk 1]
```diff
@@ -56,41 +56,34 @@ const getRouter = () => {
...
 );
-const useSelectedLayoutSegments = fn(originalNavigation.useSelectedLayoutSegments).mockName(
+export const useSelectedLayoutSegments = fn(originalNavigation.useSelectedLayoutSegments).mockName(
   'useSelectedLayoutSegments'
 );
-const useRouter = fn(originalNavigation.useRouter).mockName('useRouter');
-const useServerInsertedHTML = fn(originalNavigation.useServerInsertedHTML).mockName(
+export const useRouter = fn(originalNavigation.useRouter).mockName('next/navigation::useRouter');
+export const useServerInsertedHTML = fn(originalNavigation.useServerInsertedHTML).mockName(
   'useServerInsertedHTML'
```
Comment (yannbf) [inline code/frameworks/nextjs/src/export-mocks/navigation/index.ts:76]:
```suggestion
  'next/navigation::useServerInsertedHTML'
```

File: code/frameworks/nextjs/src/export-mocks/navigation/index.ts [hunk 2]
```diff
@@ -56,41 +56,34 @@ const getRouter = () => {
...
+);
+export const usePathname = fn(originalNavigation.usePathname).mockName(
+  'next/navigation::usePathname'
+);
+export const useSelectedLayoutSegment = fn(originalNavigation.useSelectedLayoutSegment).mockName(
   'useSelectedLayoutSegment'
 );
-const useSelectedLayoutSegments = fn(originalNavigation.useSelectedLayoutSegments).mockName(
+export const useSelectedLayoutSegments = fn(originalNavigation.useSelectedLayoutSegments).mockName(
   'useSelectedLayoutSegments'
```
Comment (yannbf) [inline code/frameworks/nextjs/src/export-mocks/navigation/index.ts:72]:
```suggestion
  'next/navigation::useSelectedLayoutSegments'
```

File: code/frameworks/nextjs/src/export-mocks/navigation/index.ts [hunk 3]
```diff
@@ -56,41 +56,34 @@ const getRouter = () => {
...
-const usePathname = fn(originalNavigation.usePathname).mockName('usePathname');
-const useSelectedLayoutSegment = fn(originalNavigation.useSelectedLayoutSegment).mockName(
+export const useSearchParams = fn(originalNavigation.useSearchParams).mockName(
+  'next/navigation::useSearchParams'
+);
+export const usePathname = fn(originalNavigation.usePathname).mockName(
+  'next/navigation::usePathname'
+);
+export const useSelectedLayoutSegment = fn(originalNavigation.useSelectedLayoutSegment).mockName(
   'useSelectedLayoutSegment'
```
Comment (yannbf) [inline code/frameworks/nextjs/src/export-mocks/navigation/index.ts:69]:
```suggestion
  'next/navigation::useSelectedLayoutSegment'
```

File: code/frameworks/nextjs/src/export-mocks/navigation/index.ts [hunk 4]
```diff
@@ -56,41 +56,34 @@ const getRouter = () => {
...
-const useServerInsertedHTML = fn(originalNavigation.useServerInsertedHTML).mockName(
+export const useRouter = fn(originalNavigation.useRouter).mockName('next/navigation::useRouter');
+export const useServerInsertedHTML = fn(originalNavigation.useServerInsertedHTML).mockName(
   'useServerInsertedHTML'
 );
-const notFound = fn(originalNavigation.notFound).mockName('notFound');
-const permanentRedirect = fn(originalNavigation.permanentRedirect).mockName('permanentRedirect');
+export const notFound = fn(originalNavigation.notFound).mockName('next/navigation::notFound');
+export const permanentRedirect = fn(originalNavigation.permanentRedirect).mockName(
+  'permanentRedirect'
```
Comment (yannbf) [inline code/frameworks/nextjs/src/export-mocks/navigation/index.ts:80]:
```suggestion
  'next/navigation::permanentRedirect'
```

File: test-storybooks/portable-stories-kitchen-sink/nextjs/stories/__snapshots__/portable-stories.test.tsx.snap [hunk 5]
```diff
@@ -794,7 +794,7 @@ exports[`renders nextHeaderStories stories renders Default 1`] = `
         </strong>
          
         <span>
-          firstName=Jane; lastName=Doe
+          firstName=Jane; ; lastName=Doe;
```
Comment (JReinhold) [inline test-storybooks/portable-stories-kitchen-sink/nextjs/stories/__snapshots__/portable-stories.test.tsx.snap:797]: What's going on here?

**Review by JReinhold — APPROVED** [review]

File: test-storybooks/portable-stories-kitchen-sink/nextjs/stories/__snapshots__/portable-stories.test.tsx.snap (hunk 5 above)
Comment (kasperpeulen) [reply test-storybooks/portable-stories-kitchen-sink/nextjs/stories/__snapshots__/portable-stories.test.tsx.snap:797]: We are using now even closer what does Next does, so I guess it must be okay.

**Pull request merged**